lib/valkyrie/persistence/fedora/query_service.rb

```diff
@@ -74,6 +74,7 @@ def content_with_inbound(id:)
     end
 
     def find_inverse_references_by(resource:, property:)
+      validate_id resource.id
```

Comment:
This is efficient, but the exception message wouldn't be very good. You'd pass a resource and it would tell you that "id must be a Valkyrie::ID"